data/migrations/V0085__fix_ofec_candidate_totals_mv.sql

```diff
@@ -0,0 +1,296 @@
+
+SET search_path = public;
+
+DROP MATERIALIZED VIEW IF EXISTS ofec_candidate_totals_mv_tmp;
+
+CREATE MATERIALIZED VIEW ofec_candidate_totals_mv_tmp AS
+ WITH totals AS (
+         SELECT ofec_totals_house_senate_mv.committee_id,
+            ofec_totals_house_senate_mv.cycle,
+            ofec_totals_house_senate_mv.receipts,
+            ofec_totals_house_senate_mv.disbursements,
+            ofec_totals_house_senate_mv.last_cash_on_hand_end_period,
+            ofec_totals_house_senate_mv.last_debts_owed_by_committee,
+            ofec_totals_house_senate_mv.coverage_start_date,
+            ofec_totals_house_senate_mv.coverage_end_date,
+            false AS federal_funds_flag
+           FROM ofec_totals_house_senate_mv
+        UNION ALL
+         SELECT ofec_totals_presidential_mv.committee_id,
+            ofec_totals_presidential_mv.cycle,
+            ofec_totals_presidential_mv.receipts,
+            ofec_totals_presidential_mv.disbursements,
+            ofec_totals_presidential_mv.last_cash_on_hand_end_period,
+            ofec_totals_presidential_mv.last_debts_owed_by_committee,
+            ofec_totals_presidential_mv.coverage_start_date,
+            ofec_totals_presidential_mv.coverage_end_date,
+            ofec_totals_presidential_mv.federal_funds_flag
+           FROM ofec_totals_presidential_mv
+ ), link AS ( --Off-year special election candidates have > 1 row/cycle
+         SELECT DISTINCT cand_id,
+            cand_election_yr + cand_election_yr % 2 as rounded_election_yr,
+            fec_election_yr,
+            cmte_id,
+            cmte_dsgn
+          FROM ofec_cand_cmte_linkage_mv
+          WHERE cmte_dsgn IN ('P', 'A')
+ ), cycle_totals AS (
+         SELECT DISTINCT ON (link.cand_id, totals.cycle) link.cand_id AS candidate_id,
+            max(link.rounded_election_yr) AS election_year,
+            totals.cycle,
+            false AS is_election,
+            sum(totals.receipts) AS receipts,
+            sum(totals.disbursements) AS disbursements,
+            (sum(totals.receipts) > (0)::numeric) AS has_raised_funds,
+            sum(totals.last_cash_on_hand_end_period) AS cash_on_hand_end_period,
+            sum(totals.last_debts_owed_by_committee) AS debts_owed_by_committee,
+            min(totals.coverage_start_date) AS coverage_start_date,
+            max(totals.coverage_end_date) AS coverage_end_date,
+            (array_agg(totals.federal_funds_flag) @> ARRAY[true]) AS federal_funds_flag
+           FROM link
+             INNER JOIN totals
+                ON link.cmte_id = totals.committee_id
+                AND link.fec_election_yr = totals.cycle
+             LEFT JOIN ofec_candidate_election_mv election
+                ON link.cand_id = election.candidate_id
+                AND totals.cycle <= election.cand_election_year
+                AND totals.cycle > election.prev_election_year
+          GROUP BY link.cand_id, election.cand_election_year, totals.cycle
+ ), election_aggregates AS (
```

Comment:
The left join here ( line 66 - 69 ) may not be necessary here because of the distinct logic used in link subquery.  And because ofec_candidate_election_mv is used on the right side of the left join, it will not have any effect on the result of cycle_totals subquery.
@vrajmohan @lbeaufort

Reply:
@hcaofec good catch, thanks for taking a look! I had to make more changes to `ofec_candidate_totals_mv` for future cycle work. Because this changeset is thoroughly tested, I will look at removing this join in https://github.com/fecgov/openFEC/pull/3219/ and re-test if that sounds ok to you.